src/DataPersister/CommandRequestDataPersister.php

```diff
@@ -28,21 +35,34 @@ public function supports($data): bool
     /**
      * {@inheritdoc}
      */
-    public function persist($data)
+    public function persist($data): object
     {
-        $this->handleCommand($data);
+        if (!$data instanceof CommandRequestInterface) {
+            throw new \LogicException(sprintf('Expected data to be "%s".', CommandRequestInterface::class));
+        }
+
+        $this->commandBus->dispatch($data->getCommand());
+
+        if ($data instanceof QueryRequestInterface) {
+            $data = $this->handle($data->getQuery());
+
+            if (null === $data) {
+                throw new \UnexpectedValueException('Query result must not be null.');
+            }
+        }
+
+        return $data;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function remove($data)
+    public function remove($data): void
     {
-        $this->handleCommand($data);
-    }
+        if (!$data instanceof CommandRequestInterface) {
+            throw new \LogicException(sprintf('Expected data to be "%s".', CommandRequestInterface::class));
+        }
 
-    private function handleCommand(CommandRequestInterface $data)
-    {
         $this->commandBus->dispatch($data->getCommand());
```

Comment:
Not sure, if it would be useful, to allow for some query here as well. I'm aware, that `DELETE` method is usually without content, but it is just a random thought

Reply:
Yes, I did the same at first... It might make sense.